homework/index.js

```diff
@@ -30,18 +30,157 @@
+      fork.textContent = selectedRepository.forks;
+    }
+    if (selectedRepository.updated_at !== null) {
+      const tr4 = createAndAppend('tr', tbody);
+      createAndAppend('td', tr4, { class: 'label', text: 'Updated :' });
+      const updated = createAndAppend('td', tr4, { text: '' });
+      updated.textContent = new Date(selectedRepository.updated_at).toLocaleString();
+    }
+  }
+
+  function clearContainer(container) {
+    while (container.hasChildNodes()) {
+      container.removeChild(container.firstChild);
+    }
+  }
+
+  function createContributions(contributors) {
+    const container = document.getElementById('container');
+    const rightDiv = createAndAppend('div', container, {
+      id: 'rightDiv',
+      class: 'right-div white_Frame',
+    });
+    createAndAppend('p', rightDiv, {
+      class: 'contributor-header',
+      text: 'Contributions :',
+    });
+    const ul = createAndAppend('ul', rightDiv, { id: 'ulList', class: 'contributor-list' });
+    contributors.forEach(element => {
+      const li = createAndAppend('li', ul, {
+        class: 'contributor-item',
+        tabindex: 0,
+      });
+      createAndAppend('img', li, {
+        src: element.avatar_url,
+        height: 48,
+        class: 'contributor-avatar',
+        alt: 'contributor',
+      });
+      const contInfoDiv = createAndAppend('div', li, { class: 'contributor-data' });
+      createAndAppend('div', contInfoDiv, { text: element.login });
+      createAndAppend('div', contInfoDiv, {
+        class: 'contributor-badge',
+        text: element.contributions,
+      });
+      li.addEventListener('click', () => {
+        window.open(element.html_url, '_blank');
+      });
+    });
+  }
+
+  function listenSelectElement(repositories, root) {
+    const container = document.getElementById('container');
+    const select = document.getElementById('select');
+    select.addEventListener('change', () => {
+      clearContainer(container);
+      const newSelectedRepository = getSelectedRepository(
+        repositories,
+        select[select.selectedIndex].text,
+      );
+      createContainer(newSelectedRepository);
+      fetchJSON(newSelectedRepository.contributors_url, (err, contributors) => {
+        if (err) {
+          createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+        } else {
+          createContributions(contributors);
+        }
+      });
+    });
+  }
+
+  function createHeader(repositories, root) {
+    const header = createAndAppend('header', root, { class: 'header' });
+    createAndAppend('p', header, { text: 'HYF Repositories' });
+    const select = createAndAppend('select', header, {
+      id: 'select',
+      class: 'repository-selector',
+    });
+    repositories
+      .map(repository => repository.name)
+      .sort((a, b) => a.localeCompare(b, { caseFirst: 'lower' }))
+      .map((element, index) => createAndAppend('option', select, { value: index, text: element }));
```

Comment:
I overlooked this in my first review: instead of sorting just an array of repository names (so that you can show those names in sorted order in the `<select>` element it is far more efficient to sort the complete array of repository objects. In that way there will be one-on-one correspondence between the index of the selected option from the `<select>` element and the index of the corresponding repository object:

```js
repositories.sort((a, b) => a.name.localeCompare(b.name));

repositories.forEach((repository, index) =>
  createAndAppend('option', select, { value: index, text: repository.name }),
);

const defaultRepository = repositories[0];
```

You now no longer need your function `getSelectedRepository` and can delete it.